view/next-project/src/components/common/Input/Input.tsx

```diff
@@ -11,23 +11,36 @@ interface Props {
   onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
   children?: React.ReactNode;
   type?: string;
+  datalist?: { id: number; name: string }[];
+  listKey?: string;
+  enableDatalist?: boolean;
 }
 
 function Input(props: Props): JSX.Element {
   const className =
     'rounded-full border border-primary-1 py-2 px-4' +
     (props.className ? ` ${props.className}` : '');
   return (
-    <input
-      className={clsx(s.input, className)}
-      placeholder={props.placeholder}
-      id={props.id}
-      value={props.value}
-      onChange={props.onChange}
-      type={props.type}
-    >
-      {props.children}
-    </input>
+    <div>
+      <input
+        className={clsx(s.input, className)}
+        placeholder={props.placeholder}
+        id={props.id}
+        value={props.value}
+        onChange={props.onChange}
+        type={props.type}
+        list={props.enableDatalist ? props.listKey : undefined}
```

Comment:
```suggestion
        list={props.datalist && props.listKey}
```
listにはpropsにdatalistがあれば、props.listkeyを渡すで十分かと思います。